python/ray/_private/runtime_env/runtime_env_agent/runtime_env_agent.py

```diff
@@ -549,12 +617,220 @@ async def GetRuntimeEnvsInfo(self, request, context):
+
+                    if parent:
+                        # Sometimes, the parent is changed to the `init` process.
+                        # In this case, the parent.pid is 1.
+                        init_assigned_for_parent = parent.pid == 1
+                        # Sometimes, the parent is dead, and the pid is reused
+                        # by other processes. In this case, this condition is triggered.
+                        parent_changed = self.ppid != parent.pid
+
+                    if parent_gone or init_assigned_for_parent or parent_changed:
+                        parent_death_cnt += 1
+                        logger.warning(
+                            f"Raylet is considered dead {parent_death_cnt} X. "
+                            f"If it reaches to {agent_consts.PARENT_DEATH_THREASHOLD},"
+                            f" the agent will kill itself. Parent: {parent}, "
+                            f"parent_gone: {parent_gone}, "
+                            f"init_assigned_for_parent: {init_assigned_for_parent}, "
+                            f"parent_changed: {parent_changed}."
+                        )
+                        if parent_death_cnt < agent_consts.PARENT_DEATH_THREASHOLD:
+                            await asyncio.sleep(agent_consts.CHECK_PARENT_INTERVAL_S)
+                            continue
+                        sys.exit(0)
+                    else:
+                        parent_death_cnt = 0
+                    await asyncio.sleep(agent_consts.CHECK_PARENT_INTERVAL_S)
+            except Exception:
+                logger.exception("Failed to check parent PID, exiting.")
+                sys.exit(1)
+
+        if sys.platform not in ["win32", "cygwin"]:
+            check_parent_task = create_task(_check_parent())
+
+        # Start a grpc asyncio server.
+        if self._grpc_server:
+            await self._grpc_server.start()
             runtime_env_agent_pb2_grpc.add_RuntimeEnvServiceServicer_to_server(
-                self, server
+                self, self._grpc_server
             )
+        # Save agent port to redis
+        await self._gcs_aio_client.internal_kv_put(
+            f"{agent_consts.RUNTIME_ENV_AGENT_PORT_PREFIX}{self._node_id}".encode(),
+            json.dumps([self._grpc_port]).encode(),
+            True,
+            namespace=ray_constants.KV_NAMESPACE_DASHBOARD,
+        )
+        # Register agent to agent manager.
+        raylet_stub = (
+            runtime_env_agent_manager_pb2_grpc.RuntimeEnvAgentManagerServiceStub(
+                self._aiogrpc_raylet_channel
+            )
+        )
+        await raylet_stub.RegisterRuntimeEnvAgent(
+            runtime_env_agent_manager_pb2.RegisterRuntimeEnvAgentRequest(
+                agent_id=self._agent_id,
+                agent_port=self._grpc_port,
+                agent_ip_address=self._node_ip_address,
+            )
+        )
+        tasks = [check_parent_task]
```

Comment:
```suggestion
        if sys.platform not in ["win32", "cygwin"]:
            tasks = [check_parent_task]
```

Do we need to add this?